contracts/VolOracle.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.4;
+
+import { Observation } from "@uniswap/v3-core/contracts/libraries/Oracle.sol";
+import { IUniswapV3Pool } from "@uniswap/v3-core/contracts/interfaces/IUniswapV3Pool.sol";
+
+contract VolOracle {
+    // constructor
+    // initializes a uniswap
+
+    // observation size
+    // stores 30 days of data (with some buffer)
+    uint256 public constant OBSERVATION_SIZE = 345600;
+    uint256 public constant UNIV3_MAX_CARDINALITY = 65535;
+    // @dev Stores Observation arrays for each pool
+    mapping(address => Observation[UNIV3_MAX_CARDINALITY]) public observations;
```

Comment:
Is the goal to copy the data from UNI? I think we may actually copy longer time period (OBSERVATION_SIZE instead of UNIV3_MAX_CARDINALITY)